pkg/framework/context/context_test.go

```diff
@@ -9,22 +9,78 @@ package context
 import (
 	"testing"
 
+	"github.com/go-openapi/runtime/middleware/denco"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api"
+
 	mocktransport "github.com/hyperledger/aries-framework-go/pkg/internal/didcomm/transport/mock"
 	"github.com/stretchr/testify/require"
 	errors "golang.org/x/xerrors"
 )
 
 func TestNewProvider(t *testing.T) {
-	prov, err := New()
-	require.NoError(t, err)
-	require.Empty(t, prov.OutboundTransport())
+	t.Run("test new with default", func(t *testing.T) {
+		prov, err := New(nil)
+		require.NoError(t, err)
+		require.Empty(t, prov.OutboundTransport())
+	})
+	t.Run("test new with outbound transport", func(t *testing.T) {
+		prov, err := New(nil, WithOutboundTransport(mocktransport.NewOutboundTransport("success")))
+		require.NoError(t, err)
+		require.NotEmpty(t, prov.OutboundTransport())
+	})
+	t.Run("test error return from options", func(t *testing.T) {
+		_, err := New(nil, func(opts *Provider) error {
+			return errors.New("error creating the framework option")
+		})
+		require.Error(t, err)
+	})
 
-	prov, err = New(WithOutboundTransport(mocktransport.NewOutboundTransport("success")))
-	require.NoError(t, err)
-	require.NotEmpty(t, prov.OutboundTransport())
+	t.Run("test new with protocol service", func(t *testing.T) {
+		newMockSvc := func(prv api.Provider) (api.ProtocolSvc, error) {
+			return mockProtocolSvc{getAPIHandlersValue: []api.Handler{mockHandler{path: "testPath"}}}, nil
```

Comment:
same comment: Use https://github.com/hyperledger/aries-framework-go/blob/master/pkg/internal/common/support/httphandler.go#L15

Reply:
done

Reply:
You forgot to push the updated file?

Reply:
done